biojava-structure/src/main/java/org/biojava/nbio/structure/SubstructureIdentifier.java

```diff
@@ -115,7 +125,11 @@ public SubstructureIdentifier(String pdbId, List<ResidueRange> ranges) {
 		if(ranges == null) {
 			throw new NullPointerException("Null ranges list");
 		}
-		this.pdbId = pdbId;
+		try {
+			this.pdbId = new PDBId(pdbId);
+		} catch (PDBIdException e) {
+			throw new IllegalArgumentException(e);
```

Comment:
Again, should either ignore the malformed Id (perhaps set `this.pdbId = null`) or declare `throws PDBIdException`.

Reply:
I declared it to throw PDBIdException